docs/development/faq.md

```diff
@@ -0,0 +1,17 @@
+# Developer FAQ
```

Comment:
I'd find this better presented as a document describing "Room DAG concepts" with headings for "Outliers", "State groups", etc, rather than a FAQ, but ymmv.

Reply:
I think I would +1 this and place it in the docs hierarchy above https://matrix-org.github.io/synapse/latest/auth_chain_difference_algorithm.html

Reply:
Thanks for the shout on a better spot to put this 🏴‍☠️